concurrency/wait_time.go

```diff
@@ -0,0 +1,16 @@
+package concurrency
+
+import "time"
+
+func EstimatedWaitTime(limiter Limiter, average time.Duration) time.Duration {
```

Comment:
Just wondering why it's `Estimated` wait time? Is it just because we aren't sure that it will work in the retry after the initial wait time (therefore maybe there will be MORE wait time)?

Reply:
It's based on the average time, therefore, there's uncertainty as to how much time it'll actually take.

e.g. Tasks typically take 2s to complete, there are 15 waiting at the moment, and there are 5 concurrent threads, therefore, if the tasks were to be enqueued now, it _should_ take about 2s / 5 * 15 = 6s before it starts being processed.